cvmfs/sync_union_tarball.h

```diff
@@ -0,0 +1,77 @@
+/**
+ * This file is part of the CernVM File System
+ *
+ * This file defines a class which derives `SyncUnion` to
+ * provide support for tarballs
+ *
+ */
+
+#ifndef CVMFS_SYNC_UNION_TARBALL_H_
+#define CVMFS_SYNC_UNION_TARBALL_H_
+
+#include "sync_union.h"
+
+#include <archive.h>
+
+#include <set>
+#include <string>
+
+#include "sync_mediator.h"
+
+struct archive;
+
+namespace publish {
+
+class SyncUnionTarball : public SyncUnion {
+ public:
+  SyncUnionTarball(AbstractSyncMediator *mediator,
+                   const std::string &rdonly_path,
+                   const std::string &union_path,
+                   const std::string &scratch_path,
+                   const std::string &tarball_path,
+                   const std::string &base_directory);
+
+  /*
+   * Delete the working directories, where the tar is being uncompressed.
+   */
+  // ~SyncUnionTarball();
+
+  /*
+   * Check that the tarball is actually valid and that can be open.
+   * Similarly it creates the base directory or check that it has the ability to
+   * write in it.
+   * Finally it untar the tarball, recursively.
```

Comment:
This last step should move into the `Traverse()` method, I think.  Let's discuss.